File: .Backend-old/SkyLink.BookingService/Models/Booking.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SkyLink.BookingService.Models
{
    public class Booking
    {
        public int Id { get; set; }

        [Required]
        public string BookingReference { get; set; } = string.Empty; // e.g. "SLK-ABC12345"

        [Required]
        public int UserId { get; set; } // Logical FK to AuthService.User

        [Required]
        public int FlightId { get; set; } // Logical FK to FlightService.Flight

        [Required]
        public DateTime BookingDate { get; set; } = DateTime.UtcNow;

        [Required]
        public string Status { get; set; } = "Pending"; // Pending, Confirmed, Cancelled

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalPrice { get; set; }

        public List<Ticket> Tickets { get; set; } = new List<Ticket>();
    }
}

File: server/models/Booking.js
const mongoose = require('mongoose');

const bookingSchema = new mongoose.Schema(
  {
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    flight: { type: mongoose.Schema.Types.ObjectId, ref: 'Flight', required: true },
    seatNumber: { type: String, required: true },
    passengerName: { type: String, required: true },
    passengerAge: { type: Number },
    bookingReference: { type: String, required: true, unique: true },
    status: { type: String, enum: ['Pending', 'Confirmed', 'Cancelled'], default: 'Confirmed' },
    totalPrice: { type: Number, required: true },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Booking', bookingSchema);

